ellama.el

```diff
@@ -214,69 +226,19 @@ in.  Default value is (current-buffer).
 	      (set-marker-insertion-type end t)
 	      (spinner-start ellama-spinner-type)
 	      (llm-chat-streaming ellama-provider
-				  (llm-make-simple-chat-prompt prompt)
+				  ellama--chat-prompt
 				  insert-text
 				  (lambda (text)
 				    (funcall insert-text text)
 				    (with-current-buffer buffer
 				      (undo-amalgamate-change-group ellama--change-group)
 				      (accept-change-group ellama--change-group)
-				      (spinner-stop)))
+				      (spinner-stop)
+				      (funcall donecb)))
 				  (lambda (_ msg)
-				    (error "Error calling the LLM: %s" msg)
-				    (cancel-change-group ellama--change-group)))))))))
```

Comment:
I'm pretty sure this ordering was a bug.

Reply:
Sure.